agent/consul/state/connect_ca_test.go

```diff
@@ -206,18 +206,18 @@ func TestStore_CARootSetList(t *testing.T) {
 	assert.True(watchFired(ws), "watch fired")
 
 	// Read it back out and verify it.
-	expected := *ca1
+
 	expected.RaftIndex = structs.RaftIndex{
 		CreateIndex: 1,
 		ModifyIndex: 1,
 	}
-
 	ws = memdb.NewWatchSet()
 	_, roots, err := s.CARoots(ws)
 	assert.Nil(err)
 	assert.Len(roots, 1)
 	actual := roots[0]
 	assert.Equal(&expected, actual)
+	assertDeepEqual(t, expected, *actual)
```

Comment:
Do we need both of these still? They seem like duplicate assertions. I guess you added the second one to get better details out? I think we could remove the first `assert.Equal`